common/components/PrintableWallet/index.tsx

```diff
@@ -1,53 +1,53 @@
-import { PaperWallet } from 'components';
 import React from 'react';
-import printElement from 'utils/printElement';
+import { PaperWallet } from 'components';
 import { stripHexPrefix } from 'libs/values';
-import translate, { translateRaw } from 'translations';
-
-export const print = (address: string, privateKey: string) => () =>
-  address &&
-  privateKey &&
-  printElement(<PaperWallet address={address} privateKey={privateKey} />, {
-    popupFeatures: {
-      scrollbars: 'no'
-    },
-    styles: `
-      * {
-        box-sizing: border-box;
-      }
-
-      body {
-        font-family: Lato, sans-serif;
-        font-size: 1rem;
-        line-height: 1.4;
-        margin: 0;
-      }
-    `
-  });
+import translate from 'translations';
 
 interface Props {
   address: string;
   privateKey: string;
 }
 
-const PrintableWallet: React.SFC<Props> = ({ address, privateKey }) => {
-  const pkey = stripHexPrefix(privateKey);
-
-  return (
-    <div>
-      <PaperWallet address={address} privateKey={pkey} />
-      <a
-        role="button"
-        aria-label={translateRaw('X_PRINT')}
-        aria-describedby="x_PrintDesc"
-        className="btn btn-lg btn-primary btn-block"
-        onClick={print(address, pkey)}
-        style={{ margin: '10px auto 0', maxWidth: '260px' }}
-      >
-        {translate('X_PRINT')}
-      </a>
-    </div>
-  );
-};
-
-export default PrintableWallet;
+interface State {
+  paperWalletImage: string;
+}
+
+export default class PrintableWallet extends React.Component<Props, State> {
+  public state: State = {
+    paperWalletImage: ''
+  };
+
+  private paperWallet: PaperWallet | null;
+
+  public componentDidMount() {
+    setTimeout(() => {
+      if (!this.paperWallet) {
+        return this.componentDidMount();
```

Comment:
What's this supposed to do?

Reply:
This _should_ never happen, but typescript complained that paperwallet could be null, so this is just in case that were to happen, it would try again in a sec. The initial setTimeout is to make sure everything  has a chance to load properly before PNGifying it (Things like <img/> tags can take a blip.)